### ISSUE TITLE
Βήμα 19α: PyTorch και νευρωνικά δίκτυα

### ISSUE THREAD

**Issue by BlackPianoCat** [slp-ntua/patrec-labs#16]
Καλησπέρα σας! Στο βήμα 19α, θέλετε να υλοποιήσουμε κάποιον έτοιμο dataloader ή θέλετε να φτιάξουμε κάποιον δικό μας;

Και γενικά εκτός από τα links που μας έχετε στην εκφώνηση μήπως υπάρχει τίποτε άλλο να μας προτείνετε για εμάς που δεν έχουμε ξαναδεί ποτέ νευρωνικά δίκτυα (γιατί μου φαίνονται αρκετά παράξενα όσα πράγματα διαβάζω);

**Comment by georgepar** [comment]
Αυτό που θέλουμε είναι να φτιάξετε μια κλάση που να κληρονομεί από την torch.util.data.Dataset η οποία θα σας γυρνάει τα δεδομένα σας σε μορφή τουπλας (sample, label) και μετά να την κάνετε wrap με ένα torch.util.data.DataLoader, ο οποίος αναλαμβάνει το batching.

Αν ακολουθήσεις τα βήματα στο Pytorch Tutorial που δίνουμε και με λίγες μικρές προσαρμογές θα είσαι οκ.

Μια καλή πηγή για να μάθεις pytorch είναι τα tutorials που παρέχει (https://pytorch.org/tutorials/). Έχει κάποια beginner tutorials που σε βάζουν στη φιλοσοφία. 

> γιατί μου φαίνονται αρκετά παράξενα όσα πράγματα διαβάζω

Συγκεκριμένα;

**Comment by BlackPianoCat** [comment]
> Συγκεκριμένα;

Οκ είδα κάποια βίντεο και διάβασα κάποια πράγματα χθες και με έβαλαν κάπως στην φιλοσοφία των neural networks, τώρα το μόνο που μένει είναι να γράψω τον κώδικα. Για την θεωρία των neural networks που χρειαζόμαστε ρώταγα βασικά. Να διαβάσω ας πούμε κάτι από το βιβλίο του Goodfellow (ποιά κεφάλαια)?

**Comment by georgepar** [comment]
Αν σου αρέσει κάτι πιο μαθηματικό δες λίγο το Κεφ. 6 από το Goodfellow:
https://www.deeplearningbook.org/contents/mlp.html

Αν θες κάτι πιο applied / διαισθητικό δες αυτο:
http://neuralnetworksanddeeplearning.com/chap1.html

Γενικά όλα αυτά θα καλυφθούν και στο μάθημα, οπότε μην αγχώνεστε.
